pkg/controller/acmechallenges/sync.go

```diff
@@ -92,8 +93,19 @@ func (c *controller) Sync(ctx context.Context, chOriginal *cmacme.Challenge) (er
 	// This finalizer ensures that the challenge is not garbage collected before
 	// cert-manager has a chance to clean up resources created for the
 	// challenge.
+	//
+	// API Transition
+	// -- we are adding cmacme.ACMELegacyFinalizer today.
+	// -- in a future version we intend to add cmacme.ACMEDomainQualifiedFinalizer instead.
```

Comment:
will we retain support for the legacy finaliser forever? this will make users who are upgrading from very old releases in future not complain at us 😄

Reply:
That's really a project decision.

As someone who sometimes goes on bleeding edge, sometimes on LTS, and sometimes LTS-minus-2, I've worn all the hats.

From my perspective writing this code, the comment is aspirational -- it definitely won't be resolved in this PR. And might never be.